### ISSUE TITLE
How can I get refined pseudo-label ?

### ISSUE THREAD

**Issue by Kreutzer-C** [xmed-lab/DFG#1]
Thanks for your outstanding work! 
How can I get the refined pseudo-label in train_target_adapt_SAM stage?
After I finished the source pre-train and FA stage, I find following two configs in `configs/train_target_adapt_SAM.yaml` is confusing:

<img width="712" height="45" alt="Image" src="https://github.com/user-attachments/assets/571ab527-700b-4e32-968e-7c10e43ea090" />

Looking forward to your reply

**Comment by zheangh** [comment]
Thanks for your interest. I am sorry for the confusion. The two paths indicate where you expect the refined pseudo-labels to be stored. They can be any empty directories on your machine. The code will generate the refined pseudo-labels and the postprocessed pseudo-labels in these two folders.

**Comment by Kreutzer-C** [comment]
Thanks a lot for your help ^_^
But during my re-implementation,  a new problem arose:
Is the .pth format model checkpoint saved at `<save_root>/<experiment_name>/saved_models` is the final best model of DFG in target domain after I finish the main_trainer_sam stage? Among them `<save_root>` and `<experiment_name>` are configured in `train_target_adapt_SAM.yaml`

**Comment by zheangh** [comment]
After each experiment at any stage, the experiment logs (saved models, TensorBoard files, etc.) will be saved under `<save_root>/<experiment_name>/`. Besides, the parameter `save_epochs` in the `train_target_adapt_SAM.yaml` config file means the model saving interval. It is set to 100, so only the model in the last epoch is saved, but its performance should be close to the best model.

**Comment by Kreutzer-C** [comment]
Thanks for your reply, but I find it difficult to reproduce the model performance reported in your paper for the MR2CT task. The table below shows my reproduction results. Strangely, the CT2MR task seems good. Is there something I might have overlooked?

<img width="654" height="200" alt="Image" src="https://github.com/user-attachments/assets/de23404e-307a-4dee-a3e9-7fed9a107d74" />

**Comment by zheangh** [comment]
I see your result after FA is 53.24, which is much lower than 73.2 in our paper (see Table IV). The problem should be with the model after FA. I have run the code for the FA stage again and got results similar to those in our paper, as follows:

<img width="594" height="401" alt="Image" src="https://github.com/user-attachments/assets/5166beef-fcfb-401c-b0d8-093a9a6a5348" />

For debugging, you can begin with using our released source model for the FA stage training, to see if the result is significantly improved. Changing hyperparameters may also be useful, because the optimal hyperparameters may be different on different machines. For example, the `eval_steps` in `train_target_adapt_FA.yaml` may be a key parameter, which influences the learning rate decay rate (see line 264 in `./trainers/target_adapt_FA_trainer.py`).

**Comment by lucky194** [comment]
@zheangh Hello! Thanks for your outstanding work!

I am reproducing the DFG paper and have successfully run the full pipeline, achieving a final model with an average Dice of 84.25% on the validation set (CHAOS MRI → BTCV CT), which is close to the reported 84.9%. The model checkpoint is named model_step_100_dice_0.8425.pth.

However, I am unable to obtain the per-organ Dice scores (spleen, left and right kidneys, liver). When I evaluate the model on the validation set, the results are very low (around 0.4). I suspect this may be because the model returns a tuple (features, seg_logits) and I might not be handling the segmentation output correctly. Also, the training log only records the pseudo-label quality (after DBS and CP), not the final validation metrics.

Could you please advise:

How should the final model be correctly evaluated to obtain per-organ Dice?

Is the validation set used during training fixed? How can I access it via code (e.g., phase='val' or a fixed split)?

Is there a recommended evaluation script?

Thank you! My environment: PyTorch 2.0.1, CUDA 11.8, RTX 3090.

**Comment by Yikmat1220** [comment]
Thanks for your wonderful work！

I have the same issue.When I reproduced the MRI2CT task,it is difficult to achieve the result from your report espetially in FA part.My average dice is about 56.

But result of the CT2MRI task is correct.

I`m very confusing now,is there any suggestion for this problem？

Thank you！
